module.go

```diff
@@ -391,3 +391,33 @@ func (m *Module) Iterator() *BPFObjectIterator {
 		prevMap:  nil,
 	}
 }
+
+func (m *Module) AttachProgs() error {
+	iters := m.Iterator()
+	for {
+		prog := iters.NextProgram()
+		if prog == nil {
+			break
+		}
+		link, err := prog.AttachGeneric()
```

Comment:
Before attempting to attach, shouldn't we check for autoload and autoattach as bpf_object__attach_skeleton does:

```c
		if (!prog->autoload || !prog->autoattach)
			continue;
```

And if the link is already set?

```
		/* if user already set the link manually, don't attempt auto-attach */
		if (*link)
			continue;
```

Reply:
I add two apis: AutoLoad and GetProg to solve this.